fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -434,6 +445,10 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         queueRestoreLocale()
         queueStorageMaintenance()
         queueNimbusFetchInForeground()
+
+        if (settings().enableFxSuggest) {
+            queueFxSuggestIngestion()
```

Comment:
I think it would!

Is it thread-safe to access `settings().enableFxSuggest` directly inside the `runIfReadyOrQueue` block in `queueInitStorageAndServices`?

Reply:
It's thread-safe from a concurrent access standpoint, but still subject to the usual race condition considerations. That said, users shouldn't be able to update their preferences before this work is triggered. If the concern is about whether Nimbus will have fetched the flag correctly by then, TBH I don't have a good answer for you there. It seems like the current or your suggested implementation would have the same Nimbus ordering concern though (if any). 

Otherwise, it's actually preferable to run property reads from `settings()` on a non-main thread anyway because those are all disk reads.

Reply:
Got it—this is all super useful to know, and it's helping me form a better mental model; thank you!

> It's thread-safe from a concurrent access standpoint, but still subject to the usual race condition considerations. [...] [I]t's actually preferable to run property reads from `settings()` on a non-main thread anyway because those are all disk reads.

This is what I was wondering about; thanks!

> If the concern is about whether Nimbus will have fetched the flag correctly by then, TBH I don't have a good answer for you there.

I was curious about this, too! For completeness, summarizing @jhugman from Slack: any experiments fetched in the current session won't be applied until the next session, so we don't have to worry about `FxNimbus.features.fxSuggest.value().enabled` changing after a network fetch in the same session.